numba/parfor.py

```diff
@@ -3731,6 +3737,17 @@ def remove_dead_parfor_recursive(parfor, lives, arg_aliases, alias_map,
     assert first_body_block > 0  # we are using 0 for init block here
     last_label = max(blocks.keys())
 
+    """
+      Previously, this statement used lives_n_aliases.  That had the effect of
+      keeping variables in the init_block alive if they aliased an array that
+      was later written to.  By using just lives to indicate which variables
+      names are live at exit of the parfor but then using alias_map for the
+      actual recursive dead code removal, we keep any writes to aliased arrays
+      alive but also allow aliasing assignments (i.e., a = b) to be eliminated
+      so long as 'a' is not written to through the variable 'a' later on.
```

Comment:
```suggestion
      so long as 'a' is not written to through the variable 'b' later on.
```
do you meant that ^ ?

Reply:
I think it is the reverse.  I've pushed a fix. Thanks for the catch @stuartarchibald .

Reply:
Great, thanks!